diff -ruN a/config.def.h b/config.def.h
--- a/config.def.h	2023-07-07 21:24:49.300140628 -0500
+++ b/config.def.h	2023-07-07 21:24:31.500139830 -0500
@@ -3,10 +3,10 @@
 /* appearance */
 static const char font[]        = "monospace:size=9";
 static const char* normbgcolor  = "#222222";
-static const char* normfgcolor  = "#cccccc";
+static const char* normfgcolor  = "#bbbbbb";
 static const char* selbgcolor   = "#555555";
-static const char* selfgcolor   = "#ffffff";
-static const char* urgbgcolor   = "#111111";
+static const char* selfgcolor   = "#bbbbbb";
+static const char* urgbgcolor   = "#000000";
 static const char* urgfgcolor   = "#cc0000";
 static const char before[]      = "<";
 static const char after[]       = ">";
diff -ruN a/tabbed.c b/tabbed.c
--- a/tabbed.c	2023-07-07 21:24:49.300140628 -0500
+++ b/tabbed.c	2023-07-07 21:24:31.500139830 -0500
@@ -152,7 +152,7 @@
 	[MapRequest] = maprequest,
 	[PropertyNotify] = propertynotify,
 };
-static int bh, obh, wx, wy, ww, wh;
+static int bh, obh, wx, wy, ww, wh, vbh;
 static unsigned int numlockmask;
 static Bool running = True, nextfocus, doinitspawn = True,
             fillagain = False, closelastclient = False,
@@ -324,20 +324,30 @@
 drawbar(void)
 {
 	XftColor *col;
-	int c, cc, fc, width;
+	int c, cc, fc, width, nbh, i;
 	char *name = NULL;
+	char tabtitle[256];
 
 	if (nclients == 0) {
 		dc.x = 0;
 		dc.w = ww;
 		XFetchName(dpy, win, &name);
 		drawtext(name ? name : "", dc.norm);
-		XCopyArea(dpy, dc.drawable, win, dc.gc, 0, 0, ww, bh, 0, 0);
+		XCopyArea(dpy, dc.drawable, win, dc.gc, 0, 0, ww, vbh, 0, 0);
 		XSync(dpy, False);
 
 		return;
 	}
 
+	nbh = nclients > 1 ? vbh : 0;
+	if (bh != nbh) {
+		bh = nbh;
+		for (i = 0; i < nclients; i++)
+			XMoveResizeWindow(dpy, clients[i]->win, 0, bh, ww, wh - bh);
+		}
+	if (bh == 0)
+		return;
+
 	width = ww;
 	cc = ww / tabwidth;
 	if (nclients > cc)
@@ -367,7 +377,9 @@
 		} else {
 			col = clients[c]->urgent ? dc.urg : dc.norm;
 		}
-		drawtext(clients[c]->name, col);
+		snprintf(tabtitle, sizeof(tabtitle), "%d: %s",
+		         c + 1, clients[c]->name);
+		drawtext(tabtitle, col);
 		dc.x += dc.w;
 		clients[c]->tabx = dc.x;
 	}
@@ -991,7 +1003,7 @@
 	screen = DefaultScreen(dpy);
 	root = RootWindow(dpy, screen);
 	initfont(font);
-	bh = dc.h = dc.font.height + 2;
+	vbh = dc.h = dc.font.height + 2;
 
 	/* init atoms */
 	wmatom[WMDelete] = XInternAtom(dpy, "WM_DELETE_WINDOW", False);
